lib/internal/test_runner/tap_stream.js

```diff
@@ -190,4 +121,100 @@ function tapEscape(input) {
   );
 }
 
+function jsToYaml(indent, name, value) {
+  if (value === null || value === undefined) {
+    return '';
+  }
+
+  if (typeof value !== 'object') {
+    const prefix = `${indent}  ${name}: `;
+
+    if (typeof value !== 'string') {
+      return `${prefix}${inspect(value, inspectOptions)}\n`;
```

Comment:
i believe this can invoke user code as well via a custom inspect symbol; is that a concern?

Reply:
I've updated the inspect logic to try serializing normally, and if an exception is thrown, try again without allowing custom inspect symbols.